content/en/blogs/cilium-as-cni.md

```diff
@@ -0,0 +1,230 @@
+---
+
+## What Is Cilium
+
+[Cilium](https://cilium.io/) is an open-source project focusing on container network. It can be deployed on container platforms to transparently secure the network connection and load balancing between application workloads, such as application containers or processes. 
+
+Running on Layer 3 and Layer 4, Cilium provides conventional network and security services. It also runs on Layer 7 to secure the use of modern application protocols, such as HTTP, gRPC, and Kafka. Cilium can be integrated into popular container orchestration frameworks such as Kubernetes and Mesos.
+
+Based on a new Linux kernel technology called eBPF, Cilium enables the dynamic insertion of powerful security visibility and control logic within Linux itself. Because eBPF runs inside the Linux kernel, Cilium security policies can be applied and updated without any changes to the application code or container configuration.
+
+![cilium-arch](/images/blogs/en/cilium-as-cni/cilium-arch.png)
+
+In this blog, I'll try to explore Cilium as the CNI (Container Network Interface) on a Kubernetes cluster. If you are interested in the performance details about Cilium, check out [CNI Benchmark: Understanding Cilium Network Performance](https://cilium.io/blog/2021/05/11/cni-benchmark).
+
+## What is KubeKey
+
+Developed in Go, KubeKey is a lightweight installer that provides an easy and flexible way to install Kubernetes and KubeSphere. With KubeKey, you can install either Kubernetes only or Kubernetes and KubeSphere both. 
+
+The scenarios where you can use KubeKey:
+
+- Install Kubernetes only.
+- Install Kubernetes and KubeSphere together using one command.
+- Scale a cluster.
+- Upgrade a cluster.
+- Install Kubernetes-related add-ons (Chart or YAML).
+
+Currently, KubeKey supports the integration of Cilium as the CNI in deploying Kubernetes clusters with KubeSphere installed, which makes the deployment process simpler and more efficient. 
+
+## Cilium as the CNI: Deploy a Kubernetes Cluster with KubeSphere Installed
+
+### System requirements
+
+Before installing Cilium, you have to ensure that your system meets the minimum requirements. For example, the Linux kernel version of your host system must be 4.9.17 and later if you run Cilium using the container image `cilium/cilium`.
+
+For more information about system requirements, check out [Cilium system requirements](https://docs.cilium.io/en/v1.10/operations/system_requirements/). 
+
+### Host preparations
+
+In this blog, let's use a Ubuntu Server 20.04.1 LTS 64bit as an example.
+
+| Name  | IP Address  | Role                 |
+| ----- | ----------- | -------------------- |
+| node1 | 192.168.2.2 | etcd, master, worker |
+
+For more information about Linux compatibility, check out [Linux Distribution Compatibility Matrix](https://docs.cilium.io/en/v1.10/operations/system_requirements/#linux-distribution-compatibility-matrix).
+
+### Install KubeKey
```

Comment:
In fact, KubeKey itself is an Installer, can we use `Download KubeKey`?

Reply:
I think we need to check this with Rui Yao?
![Xnip2021-07-19_16-59-12](https://user-images.githubusercontent.com/75110798/126133381-ada2a70b-66ee-4fbf-9f93-0bea2a1f9287.png)

Reply:
dpkg is the package management tool of Ubuntu. Do we need to add the installation of CentOS?